<!DOCTYPE html>
<meta charset="utf-8">
<style>

@import url(stack.css);

.big {
  font-size: 128px;
}

.caption {
  font-size: 96px;
  text-align: center;
  text-shadow: 0px 2px 6px rgba(0,0,0,.5);
}

</style>
<body>

<section>
<h1><u>Work in Progress</u></h1>
<h3>Gideon Dresdner</h3>
<h3>January 16, 2013</h3>
</section>

<section style="background:#ffffff url(Slide5.jpg); background-size:1000px; background-repeat:no-repeat;">
<p style="color:black; position:absolute; bottom:75px; left:40px; font-size:38px;">
    <b>--Benjamin Gross' WIP. Jan. 2, 2013</b>
</p>
</section>

<section style="background:#ffffff url(Slide5.jpg); background-size:1000px; background-repeat:no-repeat;">
<div class="highlight-box" style="left:179px; top:320px; width:560px; height:70px"></div>
</section>

<section>
<h1 >CN Pipeline</h1>
<p class="caption" style="margin:40px">CBS &rarr; GISTIC</p>
<ul>
    <li>Circular Binary Segmentation (CBS)</li>
    <li>Genomic Identification of Significant Targets in Cancer (GISTIC)</li>
</ul>
</section>

<section>

<h3>Why segmentation?</h3>
<p>Noise in probe level data from SNP and CGH data.</p>

<a href="cbs_out_plot.png">cbs output plot</a> (genome location vs. log ratio)

<p>There are segments here, but where are they?  And how can we automate this?</p>

</section>

<section>
<h3>Why GISTIC?</h3>
<p>How do we find the "significant" <b>segments</b> in a population?</p>
<p>How do we find the "significant" <b>genes</b> in a population?</p>
</section>

<section>
<u><b>C</b>ircular</u> / <u><b>B</b>inary <b>S</b>egmentation</u>

<p>Looking for breakpoints by taking MLE on a list of Gaussian distributions.</p>

<ul>
    <li>Binary Segmentation: Searches for <b>single</b> breakpoints (<a
        href="http://projecteuclid.org/euclid.aos/1176343001">1975</a>)</li>
    <br/>
    <li> Circular: Searches for <b>pairs</b> of breakpoints (a circle ends where it starts)</li>
</ul>
</section>

<section>
<h1>GISTIC</h1>
...say something about the algorithm.  Quote some its input.
</section>

<section>
<h1>TODO:</h1>
<ol>
    <li>** install and run algorithms individually **</li>
    <li>pipe together</li>
    <li> <a href="http://galaxyproject.org/">Galaxy webserver</a> wrapper(s)</li>
</ol>
</section>

<section>
Successfully forked over BIC's Agilent 244K pipeline.
<a href="cbs_out_plot.png">cbs output plot</a>.

(image of a fork?)
</section>

<section>
<h1>AffySNP6.0 Surprise: step 1 is not easy &#9785</h1>
</section>

<section>
<h1 style="float:left;">Why?<br/><br/>>CBS</h1>

<div style="position: relative; top:-80px; float:right; font-size:40px; margin-top:100px;">
    <u>TCGA Level_2 Data:</u>
    <pre>probe id    signal
CN_473963   1.492
CN_473964   1.906
</pre>
    <u>CBS input:</u>
    <pre>probe id | chr | pos | signal(s)</pre>
</section>

<section>
<h1 style="float:left;">Why?<br/><br/>>GISTIC</h1>

<div style="position: relative; top:-80px; float:right; font-size:40px; margin-top:100px; margin-right:100px;">
    <u>GISTIC markerfile:</u>
<pre>#name       chr position
CN_473963   1   61735
CN_473964   1   61808
</pre>
</div>
</section>

<section>
<h1>In Progress</h1>
<ul>
    <li>Well, if the Broad can do it...</li>
    <li><code>map_probes.py</code></li>
    <li>access to Lowe's Galaxy server</li>
</ul>
</section>

<section>
<h1><code>map_probes.py</code></h1><pre>
     (chromosome, position) &rlarr; probe id
e.g. (1, 123456)            &rlarr; CN_111111
</pre>
</section>

<section>
<h1 style="margin-top:0px;">D3 Oncoprints w/ clinical data (sorting on client)</h1>
<ol>
    <li>Overview</li>
    <li>Demo</li>
    <li>d3.js</li>
    <li>Oncoprint Future</li>
    <li>Portal Future</li>
</ol>
</section>

<section>
<h1>Overview</h1>
<ul>
    <li>backend: refactoring</li>
    <li>frontend: rewrite in d3</li>
</ul>
</section>

<script src="d3.v2.js?2.7.3"></script>
<script src="stack.v0.js"></script>
